cli/lsp/documents.rs

```diff
@@ -1162,28 +1151,32 @@ impl Documents {
 
   pub fn update_config(
     &mut self,
+    root_dirs: Vec<Url>,
     maybe_import_map: Option<Arc<import_map::ImportMap>>,
     maybe_config_file: Option<&ConfigFile>,
     maybe_package_json: Option<&PackageJson>,
     npm_registry_api: NpmRegistryApi,
     npm_resolution: NpmResolution,
   ) {
     fn calculate_resolver_config_hash(
+      root_dirs: &[Url],
       maybe_import_map: Option<&import_map::ImportMap>,
       maybe_jsx_config: Option<&JsxImportSourceConfig>,
       maybe_package_json_deps: Option<&BTreeMap<String, NpmPackageReq>>,
     ) -> u64 {
       let mut hasher = FastInsecureHasher::default();
+      hasher.write_hashable(&{
+        // ensure these are sorted
+        let mut root_dirs = root_dirs.to_vec();
+        root_dirs.sort();
```

Comment:
**question:** Do we know this set vector to not contain duplicates? If so `sort_unstable` might offer somewhat better performance and avoids allocations. https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable

> It is typically faster than stable sorting, except in a few special cases, e.g., when the slice consists of several concatenated sorted sequences.

That special case does sound like this might hit that, though :)